### ISSUE TITLE
Do not generate LSTM op by incorrect param check logic

### ISSUE THREAD

**Issue by jyoungyun** [tensorflow/tensorflow#21473]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: Yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux Ubuntu 16.04
- **Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device**: x
- **TensorFlow installed from (source or binary)**: source
- **TensorFlow version (use command below)**: r1.10
- **Python version**: 3.6.5
- **Bazel version (if compiling from source)**: 0.16.0
- **GCC/Compiler version (if compiling from source)**: 7.3.0
- **CUDA/cuDNN version**: 9.2
- **GPU model and memory**: 
- **Exact command to reproduce**:

### Describe the problem
**Related to input param of state forget gate activation**

I made a simple RNN test code with below 2 lines. I attached the full code as a file. [rnn.py.txt](https://github.com/tensorflow/tensorflow/files/2269704/rnn.py.txt)
```
lstm_layer=tf.nn.rnn_cell.BasicLSTMCell(num_units)
outputs,_=tf.nn.static_rnn(lstm_layer,input,dtype=tf.float32)
```
And I tried to generate tflite file from frozen pb via toco. But it failed to generate LSTM operation because `state_forget_sig`(State forget gate activation) did not have `Split` as an input parameter. Instead it had a `Add` operation as an input and the `Add` operation had a `Split` operation. It generates the same graph when I tried to use `dynamic_rnn`. I attached the tensorboard image. 

![state_forget_sig](https://user-images.githubusercontent.com/7223627/43870358-09d57a3c-9bb2-11e8-9d7d-2f6dcd8dd5ab.png)

I fixed some codes and it succesfully generated LSTM operations.
```
Number of all operator types: 5
	EXPAND_DIMS                           :    2 	 (instrs: ???)
	CONCATENATION                         :    2 	 (instrs: 0)
	CUSTOM(Fill)                          :    2 	 (instrs: ???)
	LSTM                                  :   28 	 (instrs: ???)
	FULLY_CONNECTED                       :    1 	 (instrs: 7,307,045,943,451,530,714)
Number of all operators                       :   35 	 (total instrs: 7,307,045,943,451,530,714)
```

If I understand it correctly, I will make a PR soon.

**Comment by lemmaa** [comment]
> ... I attached the tensorboard image. state_forget_sig

@jyoungyun , the attached image was broken. Would you please update it?

**Comment by jyoungyun** [comment]
@lemmaa Thanks for letting me know. I updated image now.

**Comment by drpngx** [comment]
So, here are your options, in decreasing order of preference:

1. Have a small `C++` test that generates the binary, run the conversion, and checks that the output does indeed have an LSTM.
2. Same, but where the binary is checked in into `testdata`, with clear instructions of how to re-create it manually.
3. Check in the binary pb, and run the toco converter in an `sh_test`. Do this only in the last resort.

**Comment by jyoungyun** [comment]
@drpngx Thanks for guideline. I will add a test soon.

**Comment by aselle** [comment]
@jyoungyun , this is really awesome. Thanks for working on this.

**Comment by tensorflowbutler** [comment]
Nagging Assignees @miaout17, @aselle: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by jyoungyun** [comment]
I will update the test case for this patch asap. Thesedays I was too busy to implement test case. Sorry and I will update it soon.

**Comment by jyoungyun** [comment]
I added lstm_test.py file into tensorflow/contrib/lite/testing directoy. This file will check whether LSTM operator is generated or not.

**Comment by tensorflowbutler** [comment]
Nagging Assignees @miaout17, @aselle: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignees @miaout17, @aselle: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignees @miaout17, @aselle: It has been 44 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignees @miaout17, @aselle: It has been 59 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignees @miaout17, @aselle: It has been 77 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by jdduke** [comment]
We're working on some usability updates for LSTMs. Stay tuned.
